lua/tardis/screens/cl_screen_music.lua

```diff
@@ -1,43 +1,265 @@
-	label:DoLayout()
-
-	local text = vgui.Create( "DTextEntry3D2D", frame )
-	text.is3D2D = screen.is3D2D
-	text.sub3D2D = label:GetText()
-	text:SetFont(TARDIS:GetScreenFont(screen, "Default"))
-	text:SetSize( frame:GetWide()*0.5, frame:GetTall()*0.1 )
-	text:SetPos(frame:GetWide()*0.5 - text:GetWide()*0.5,frame:GetTall()*0.5 - text:GetTall()*0.5)
-	text.OnEnter = function()
-		ext:PlayMusic(text:GetValue())
-	end
-	
-	local stop=vgui.Create("DButton",frame)
-	stop:SetVisible(false)
-	stop:SetSize( frame:GetWide()*0.2, frame:GetTall()*0.15 )
-	stop:SetPos(frame:GetWide()*0.5 - stop:GetWide()*0.5,frame:GetTall()*0.7 - stop:GetTall()*0.5)
-	stop:SetText("Stop")
-	stop:SetFont(TARDIS:GetScreenFont(screen, "Default"))
-	stop.DoClick = function()
-		ext:StopMusic()
-	end
-	
-	frame.Think = function()
+local default_music={
+	{"Main Theme (2005)", "theme1"},
+	{"Main Theme (2009)", "theme2"},
+	{"Main Theme (2010)", "theme3"},
+	{"Main Theme (2013)", "theme4"},
+	{"Main Theme (2014)", "theme-2014"},
+	{"Ninth Doctor", "nine"},
+	{"Tenth Doctor", "ten"},
+	{"Eleventh Doctor", "eleven"},
+	{"Twelfth Doctor", "twelve"},
+	{"Thirteenth Doctor", "thirteen"},
+	{"Rose Tyler", "rose"},
+	{"Martha Jones", "martha"},
+	{"Donna Noble", "donna"},
+	{"Amy Pond", "amy"},
+	{"River Song", "river"},
+	{"Clara Oswald", "clara"},
+	{"Amy in the TARDIS", "amyinthetardis"},
+	{"Abigail's Song", "abigail"},
+	{"Not a war", "notawar"},
+	{"This is Gallifrey", "thisisgallifrey"},
+	{"Gallifrey", "gallifrey"},
+	{"Life among the distant stars", "lifeamongthedistantstars"},
+	{"Vale Decem", "valedecem"},
+	{"The Majestic Tale", "majestictale"},
+	{"Forgiven", "forgiven"},
+	{"I need to know", "ineedtoknow"},
+	{"The Wedding of River Song", "weddingofriversong"},
+	{"Together or not at all", "togetherornotatall"},
+	{"All the Strange Creatures", "allthestrangecreatures"},
+	{"You're Fired", "yourefired"},
+	{"Shepard's Boy", "shepherdsboy"},
+	{"My beautiful Ghost Monument", "mybeautifulghostmonument"},
+	{"Whose Enigma", "whoseenigma"},
+	{"The Long Song", "thelongsong"},
+	{"Infinite Potential", "infinitepotential"},
+	{"The New Doctor", "thenewdoctor"},
+	{"Down to Earth", "downtoearth"},
+	{"Sonic Screwdriver", "sonicscrewdriver"},
+	{"My Husband's Home", "myhusbandshome"},
+	{"Doomsday", "doomsday"},
+	{"Dark and Endless Dalek Night", "darkandendlessdaleknight"},
+	{"Corridors anf Fire Escape", "corridorsandfireescape"},
+	{"The Greatest Story Never Told", "greateststorynevertold"},
+
+}
+
+--Custom music
+
+local custom_music
+
+local filename = "tardis2_custom_music.txt"
+if file.Exists(filename,"DATA") then
+	custom_music = TARDIS.von.deserialize(file.Read(filename,"DATA"))
+else
+	custom_music = {}
+end
+
+function TARDIS:SaveCustomMusic()
+	file.Write(filename, TARDIS.von.serialize(custom_music))
+end
+
+function TARDIS:AddCustomMusic(name, url)
+	local next = table.Count(custom_music) + 1
+	custom_music[next] = {name, url}
```

Comment:
I would use table.insert here rather than manually counting the table and adding it to the +1: https://wiki.facepunch.com/gmod/table.insert

Reply:
makes sense
I just suggested that way since it was being used at "destination" screen